File: packages/sdk/java/src/main/java/com/ihui/ai/sdk/module/UserApi.java
package com.ihui.ai.sdk.module;

import com.fasterxml.jackson.databind.JsonNode;
import com.ihui.ai.sdk.BaseClient;
import com.ihui.ai.sdk.model.UsageResponse;
import com.ihui.ai.sdk.model.WorkflowRequest;

/**
 * 用户 / 工作区 / 工作流 / 统计模块。
 *
 * <p>端点(9 个):
 * <ul>
 *   <li>GET  /v1/me(当前用户 + 配额)</li>
 *   <li>GET  /v1/projects(项目列表)</li>
 *   <li>GET  /v1/projects/:id/files(项目文件)</li>
 *   <li>GET  /v1/workflows/:id(工作流详情)</li>
 *   <li>POST /v1/workflows/instances(运行工作流)</li>
 *   <li>POST /v1/workflows/coze/run(Coze 工作流)</li>
 *   <li>POST /v1/workflows/n8n/run(n8n 工作流)</li>
 *   <li>GET  /v1/usage(用量统计)</li>
 *   <li>GET  /v1/usage/:vendor(厂商用量)</li>
 * </ul>
 */
public final class UserApi {

    private final BaseClient client;

    /**
     * 构造 UserApi。
     *
     * @param client 底层 BaseClient
     */
    public UserApi(BaseClient client) {
        this.client = client;
    }

    /**
     * GET /v1/me(当前用户信息 + 配额)。
     *
     * @return 用户信息(JsonNode)
     */
    public JsonNode me() {
        return client.request("GET", "/me", null, JsonNode.class);
    }

    /**
     * GET /v1/projects(项目列表)。
     *
     * @return 项目列表(JsonNode)
     */
    public JsonNode listProjects() {
        return client.request("GET", "/projects", null, JsonNode.class);
    }

    /**
     * GET /v1/projects/:id/files(项目文件列表)。
     *
     * @param projectId 项目 ID
     * @return 文件列表(JsonNode)
     */
    public JsonNode listProjectFiles(String projectId) {
        return client.request("GET", "/projects/" + BaseClient.encode(projectId) + "/files",
                null, JsonNode.class);
    }

    /**
     * GET /v1/workflows/:id(工作流详情)。
     *
     * @param id 工作流 ID
     * @return 工作流详情(JsonNode)
     */
    public JsonNode getWorkflow(String id) {
        return client.request("GET", "/workflows/" + BaseClient.encode(id),
                null, JsonNode.class);
    }

    /**
     * POST /v1/workflows/instances(运行工作流)。
     *
     * @param req 工作流运行请求
     * @return 运行结果(JsonNode)
     */
    public JsonNode runWorkflow(WorkflowRequest req) {
        return client.request("POST", "/workflows/instances", req, JsonNode.class);
    }

    /**
     * POST /v1/workflows/coze/run(Coze 工作流)。
     *
     * @param req 工作流运行请求
     * @return 运行结果(JsonNode,透传上游 Coze 响应)
     */
    public JsonNode runCozeWorkflow(WorkflowRequest req) {
        return client.request("POST", "/workflows/coze/run", req, JsonNode.class);
    }

    /**
     * POST /v1/workflows/n8n/run(n8n 工作流)。
     *
     * @param req 工作流运行请求
     * @return 运行结果(JsonNode,透传上游 n8n 响应)
     */
    public JsonNode runN8nWorkflow(WorkflowRequest req) {
        return client.request("POST", "/workflows/n8n/run", req, JsonNode.class);
    }

    /**
     * GET /v1/usage(用量统计)。
     *
     * @return 用量统计
     */
    public UsageResponse getUsage() {
        return client.request("GET", "/usage", null, UsageResponse.class);
    }

    /**
     * GET /v1/usage/:vendor(厂商用量)。
     *
     * @param vendor 厂商标识
     * @return 厂商用量(JsonNode)
     */
    public JsonNode getVendorUsage(String vendor) {
        return client.request("GET", "/usage/" + BaseClient.encode(vendor),
                null, JsonNode.class);
    }
}

File: packages/sdk/dotnet/Modules/UserApi.cs
using System.Text.Json;

namespace Ihui.AI;

/// <summary>
/// 用户 / 工作区 / 工作流 / 统计模块。
/// </summary>
/// <remarks>
/// 端点(9 个):
/// <list type="bullet">
///   <item>GET  /v1/me(当前用户 + 配额)</item>
///   <item>GET  /v1/projects(项目列表)</item>
///   <item>GET  /v1/projects/:id/files(项目文件)</item>
///   <item>GET  /v1/workflows/:id(工作流详情)</item>
///   <item>POST /v1/workflows/instances(运行工作流)</item>
///   <item>POST /v1/workflows/coze/run(Coze 工作流)</item>
///   <item>POST /v1/workflows/n8n/run(n8n 工作流)</item>
///   <item>GET  /v1/usage(用量统计)</item>
///   <item>GET  /v1/usage/:vendor(厂商用量)</item>
/// </list>
/// </remarks>
public sealed class UserApi
{
    private readonly BaseClient _client;

    internal UserApi(BaseClient client) => _client = client;

    /// <summary>GET /v1/me(当前用户信息 + 配额)。</summary>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>用户信息(JsonElement)</returns>
    public Task<JsonElement?> MeAsync(CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("GET", "/me", null, cancellationToken);

    /// <summary>GET /v1/projects(项目列表)。</summary>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>项目列表(JsonElement)</returns>
    public Task<JsonElement?> ListProjectsAsync(CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("GET", "/projects", null, cancellationToken);

    /// <summary>GET /v1/projects/:id/files(项目文件列表)。</summary>
    /// <param name="projectId">项目 ID</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>文件列表(JsonElement)</returns>
    public Task<JsonElement?> ListProjectFilesAsync(string projectId, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("GET", "/projects/" + BaseClient.Encode(projectId) + "/files", null, cancellationToken);

    /// <summary>GET /v1/workflows/:id(工作流详情)。</summary>
    /// <param name="id">工作流 ID</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>工作流详情(JsonElement)</returns>
    public Task<JsonElement?> GetWorkflowAsync(string id, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("GET", "/workflows/" + BaseClient.Encode(id), null, cancellationToken);

    /// <summary>POST /v1/workflows/instances(运行工作流)。</summary>
    /// <param name="req">工作流运行请求</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>运行结果(JsonElement)</returns>
    public Task<JsonElement?> RunWorkflowAsync(WorkflowRequest req, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("POST", "/workflows/instances", req, cancellationToken);

    /// <summary>POST /v1/workflows/coze/run(Coze 工作流)。</summary>
    /// <param name="req">工作流运行请求</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>运行结果(JsonElement,透传上游 Coze 响应)</returns>
    public Task<JsonElement?> RunCozeWorkflowAsync(WorkflowRequest req, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("POST", "/workflows/coze/run", req, cancellationToken);

    /// <summary>POST /v1/workflows/n8n/run(n8n 工作流)。</summary>
    /// <param name="req">工作流运行请求</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>运行结果(JsonElement,透传上游 n8n 响应)</returns>
    public Task<JsonElement?> RunN8nWorkflowAsync(WorkflowRequest req, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("POST", "/workflows/n8n/run", req, cancellationToken);

    /// <summary>GET /v1/usage(用量统计)。</summary>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>用量统计</returns>
    public Task<UsageResponse?> GetUsageAsync(CancellationToken cancellationToken = default)
        => _client.RequestAsync<UsageResponse>("GET", "/usage", null, cancellationToken);

    /// <summary>GET /v1/usage/:vendor(厂商用量)。</summary>
    /// <param name="vendor">厂商标识</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>厂商用量(JsonElement)</returns>
    public Task<JsonElement?> GetVendorUsageAsync(string vendor, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("GET", "/usage/" + BaseClient.Encode(vendor), null, cancellationToken);
}
